### ISSUE TITLE
Suddenly connection to FB 7590 can't be established anymore

### ISSUE THREAD

**Issue by ceysa75** [dnknth/node-red-contrib-fritzapi#29]
Without having changed anything I realized this morning that my heating schedules were not set (thermostats dect 301). Looking at my node red thermostats nodes it says login failed. As already said, I have not changed any setting in the last months. I should mention here that I have updated FRITZ!OS to 07.29 a few days ago.
Of course I have now checked all entries in the configuration node (username / password), however, all seems to be right.
BTW:  Within the Fritzbox UI event list it says wrong password (sth like _connection could not be established due to wrong password_).

FRITZ!OS: 07.29
NR: 2.13 (standard installation on a raspberry, no docker)
node-red-contrib-fritzapi: 0.3.2

I appreciate you help...

Regards

Ceysa

**Comment by dnknth** [comment]
Hello @ceysa75,

No clue, honestly. "It stopped working suddenly" is a bit vague to begin with.
I use the same FB version without any issues. I would probably try to set up a new FB user with home automation access and see whether reconfiguring the nodes solves it. Good luck!

**Comment by ceysa75** [comment]
Hi dnknth,

I tried that already, ne FB user with full access, but no success. I have further investigated what the problem could be. It is as simple as the username and the password in the configuration node (tried that for different users) won't get accepted. Whereas with the identical username and password accessing the FB via Web-Gui is not a problem at all...

![image](https://user-images.githubusercontent.com/37417641/142777865-d866f93d-7af8-4ac8-ba6f-75811706526f.png)

![image](https://user-images.githubusercontent.com/37417641/142777880-cbb437c4-27b1-4a3e-9539-88f6d90a85ce.png)

![image](https://user-images.githubusercontent.com/37417641/142778036-dbf8c9b7-addb-4757-ab73-ec82f813178d.png)

Thank you...

**Comment by ceysa75** [comment]
Hi dnknth,

while trying to find the error source I had the idea to try a different node (but keeping all other parameters identical). I found the node-red-contrib-fritz node [https://flows.nodered.org/node/node-red-contrib-fritz](url)

To my great surprise: The connection is established and all is working fine...

![image](https://user-images.githubusercontent.com/37417641/142779711-813ecb22-f2b7-4a0b-8790-e92c21bce1dc.png)

Summing up: My problems are obviously not due to the user on the FB and not the recent FB firmware. So what could be the reason that the username / password of the node-red-contrib-fritzapi is not accepted by the Fritzbox ?? Really irritating...

**Comment by ceysa75** [comment]
Hi dnknth,

I solved it. I had to delete the configuration node and then generate it again (I wanted to avoid that all the time). However, now it works again. We can close this. However, what still remains is the question why this happened at all ? 

Regards

Ceysa

**Comment by ceysa75** [comment]
Unfortunately I have to re-open that issue. After re-starting node red and rebooting my raspberry it happens again. The configuration node is **not** able to login into the Fritzbox.

![image](https://user-images.githubusercontent.com/37417641/142826452-eb631410-956b-47cc-ae49-7e070992a123.png)
  
This situation makes me crazy...

**Comment by dnknth** [comment]
No clue what might be causing this. The node does not tamper with credentials, they are stored like any others. Note that `note-red-contrib-fritz` uses the TR064 API, hence the different port (49000 / 49443), whereas `node-red-contrib-fritzapi` uses the [home automation API](https://avm.de/fileadmin/user_upload/Global/Service/Schnittstellen/AHA-HTTP-Interface.pdf).

You could try to set up a dedicated user for Node-RED. Make sure to enable home automation access, this works for me.

Other than that, I have no further idea what to try, but am quite confident that the issue is not caused by `node-red-contrib-fritzapi`.

**Comment by dnknth** [comment]
Can we close this issue?

**Comment by TheFoxy** [comment]
Hi there!
Same Problem here! 
Fritzbox 7530  last update!
New Raspberry 4 with note red  an last node-red-contrib-fritzapi 0.3.2  installation.
Old version node-red-contrib-fritzapi 0.2.14 same problem.
 Admin user fritzbox with password and new user with password will not login!
 year ago node red with  0.2.14 works fine.
Fritzbox has updated 2 times in the last year!
Any   ideas?

**Comment by ceysa75** [comment]
I am still investigating. For many days no issues observed. But I have to mention that I have considerably reduced the number of nodes of this contrib to around 20 (was around 60 before).

**Comment by TheFoxy** [comment]
the fritzapi with javascript has the same error!

**Comment by TheFoxy** [comment]
Unhandled rejection 0000000000000000

**Comment by TheFoxy** [comment]
Important: Change to FRITZ!Box login process

Starting with FRITZ!OS 7.25, there is a change when logging in with the FRITZ!Box password, which app developers will need to take into account. The change is also part of the FRITZ!OS 7.24 Lab version.
look at https://avm.de/service/schnittstellen/

**Comment by dnknth** [comment]
Closing this due to lack of updates. Feel free to re-open if the issue is not yet solved.

**Comment by ceysa75** [comment]
Sorry for being late with my respond. No issues observed for a longer period. Regards and thanks for your efforts...
